homeassistant/components/climate/dyson.py

```diff
@@ -0,0 +1,194 @@
+
+STATE_DIFFUSE = "Diffuse Mode"
+STATE_FOCUS = "Focus Mode"
+FAN_LIST = [STATE_FOCUS, STATE_DIFFUSE]
+OPERATION_LIST = [STATE_HEAT, STATE_COOL]
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+                 | SUPPORT_OPERATION_MODE)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Dyson fan components."""
+    if discovery_info is None:
+        return
+
+    from libpurecoollink.dyson_pure_hotcool_link import DysonPureHotCoolLink
+    # Get Dyson Devices from parent component.
+    add_devices(
+        [DysonPureHotCoolLinkDevice(device)
+         for device in hass.data[DYSON_DEVICES]
+         if isinstance(device, DysonPureHotCoolLink)]
+    )
+
+
+class DysonPureHotCoolLinkDevice(ClimateDevice):
+    """Representation of a Dyson climate fan."""
+
+    def __init__(self, device):
+        """Initialize the fan."""
+        self._device = device
+        self._current_temp = None
+        self._pending_target_temp = None
+
+    async def async_added_to_hass(self):
+        """Callback when entity is added to hass."""
+        self.hass.async_add_job(self._device.add_message_listener,
+                                self.on_message)
+
+    def on_message(self, message):
+        """Called when new messages received from the climate."""
+        from libpurecoollink.dyson_pure_state import DysonPureHotCoolState
+
+        if isinstance(message, DysonPureHotCoolState):
+            from libpurecoollink.const import HeatMode
+            _LOGGER.debug("Message received for fan device %s : %s", self.name,
+                          message)
+            self.schedule_update_ha_state()
+            if (self._device.state.heat_mode == HeatMode.HEAT_ON.value
+                    and self._pending_target_temp is not None):
+                # Set any pending new target temperature on dyson device.
+                self._set_temperature_on_device()
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def name(self):
+        """Return the display name of this climate."""
+        return self._device.name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        if self._device.environmental_state:
+            temperature_kelvin = self._device.environmental_state.temperature
+            if temperature_kelvin != 0:
+                self._current_temp = float("{0:.1f}".format(
+                    temperature_kelvin - 273))
+        return self._current_temp
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        if self._pending_target_temp is None:
```

Comment:
Yes, because the device ignores set `target temp` request when it is in **cool mode**. Without it, setting `target temp` when the device is not in **heat mode** will render the user interface and actual device `target temp` out of sync. It also won't have a callback update (as none of the internal state changes) so this out of sync remains indefinitely. The user set values will be rendered useless as a result.

Reply:
IMO, if device ignores set target temp in some condition, you should simply log error and do nothing. as long as you always return device's target temp as entity's target temp, I cannot see the reason for out of sync.